.github/workflows/main.yml

```diff
@@ -0,0 +1,273 @@
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: sudo apt -y install libcurl4-openssl-dev libssl-dev pkg-config
+
+      - name: Install pip requirements
+        run: |
+          mkdir ./.cache && mkdir ./.cache/pip
+          chown -R $(whoami) ./.cache/pip
+          pip install -U pip setuptools wheel --cache-dir ./.cache/pip
+          pip install -r requirements/main.txt --cache-dir ./.cache/pip
+          pip install -r requirements/tests.txt --cache-dir ./.cache/pip
+          pip install -r requirements/deploy.txt --cache-dir ./.cache/pip
+          
+      - name: Setup node
+        run: |
+          curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
+          export NVM_DIR="$HOME/.nvm" 
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+          [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
+          nvm install $NODE_VERSION
+          npm i -g npm
+          npm install -g gulp-cli
+          npm ci
+
+      - name: Run tests
+        run: |
+          python -m coverage run -m pytest --strict --postgresql-host localhost
+          python -m coverage html
+          python -m coverage report -m --fail-under 100
+        env:
+          # The hostname used to communicate with the PostgreSQL service container
+          POSTGRES_HOST: localhost
+          # The default PostgreSQL port
+          POSTGRES_PORT: 5432
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Lint
+        run: |
+          sudo apt-get update && sudo apt-get install make
+          make lint ACTIONS=true
+
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Documentation
+        run: |
+          sudo apt-get update && sudo apt-get install make
+          make docs
+  
+  deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Cache pip dependencies
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-pip-deps-warehouse-p
```

Comment:
What's with the `-p` on the end of this?

Reply:
TLDR: It shouldn't be there.

The caching action is strange since it doesn't update the cache after the first time it caches dependencies, you have to change the key if you want a new one. So, I was sort of mindlessly adding characters whenever I would change stuff lol. However, I just found that it's recommended to [hash the dependency files](https://docs.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#using-contexts-to-create-cache-keys) such that the keys change when the dependencies change, so I'm gonna update the .yml file to reflect that.

Reply:
Actually, it seems I'm already hashing the dependency files where I need to be. I think I did this because I was changing which directories were being cached (e.g. `~/.cache/pip` vs `./.cache/pip`, `~/.npm` vs `~/.npm && **/node_modules`) and needed the updated cache to reflect that. Still, I'll go ahead and change the `cache-name` to be something readable.